packages/yarnpkg-fslib/sources/index.ts

```diff
@@ -284,6 +285,19 @@ export function patchFs(patchedFs: typeof fs, fakeFs: FakeFS<NativePath>): void
       // `fs.promises.realpath` doesn't have a `native` property
     }
   }
+
+  /** util.promisify implementations */
+  {
+    // Override the promisified version of `fs.read` to return an object as per
+    // https://github.com/nodejs/node/blob/dc79f3f37caf6f25b8efee4623bec31e2c20f595/lib/fs.js#L559-L560
+    // and
+    // https://github.com/nodejs/node/blob/ba684805b6c0eded76e5cd89ee00328ac7a59365/lib/internal/util.js#L293
+    // @ts-expect-error
+    patchedFs.read[promisify.custom] = async (p: number, buffer: Buffer, ...args: Array<any>) => {
```

Comment:
Shouldn't we do the same for `write` / `readv` / `writev`?

Reply:
Those other three aren't patched, so this problem doesn't occur.

If we wanted to patch those methods, then yeah we should do the promisify patching.